File: BackEnd/Service/Models/CvModel.cs
using Data.Entities;

namespace Service.Models
{
    public class CvModel
    {
        public Guid Cvid { get; set; }

        public Guid CandidateId { get; set; }

        public string? CvPdf { get; set; }

        public string CvName { get; set; }

        public string AboutMe { get; set; } = null!;


        public bool IsDeleted { get; set; } = false;

        public bool IsDefault { get; set; } = false;

        public ICollection<ApplicationModel> Applications { get; set; } 

        public CandidateModel Candidate { get; set; } = null!;

        public ICollection<CvHasSkillModel> CvHasSkills { get; set; } = null!;
    }
}

File: FrontEnd/src/app/data/cv/cv.model.ts
import { Application } from "../application/application.model";
import { Candidate } from "../candidate/candidate.model";
import { CvHasSkill } from "../cvHasSkill/cv-has-skill.model";
import { Skill } from "../skill/skill.model";

export class CV {
	cvid?: string;
	candidateId?: string;
	candidate?: Candidate;
	cvPdf?: string;
	cvName?: string;
	aboutMe?: string;
	isDeleted?: boolean;
	isDefault?: boolean;
	applications?: Application[];
	skills?: Skill[];
	cvHasSkills?: CvHasSkill[];
}
